core/include/prometheus/registry.h

```diff
@@ -52,19 +55,44 @@ class Registry : public Collectable {
 
   template <typename T>
   Family<T>& Add(const std::string& name, const std::string& help,
-                 const std::map<std::string, std::string>& labels);
+                 const std::map<std::string, std::string>& labels,
+                 std::vector<std::unique_ptr<Family<T>>>& families);
+
+  Family<Counter>& AddCounter(const std::string& name, const std::string& help,
+                              const std::map<std::string, std::string>& labels);
+  Family<Gauge>& AddGauge(const std::string& name, const std::string& help,
+                          const std::map<std::string, std::string>& labels);
+  Family<Histogram>& AddHistogram(
+      const std::string& name, const std::string& help,
+      const std::map<std::string, std::string>& labels);
+  Family<Summary>& AddSummary(const std::string& name, const std::string& help,
+                              const std::map<std::string, std::string>& labels);
 
-  std::vector<std::unique_ptr<Collectable>> collectables_;
+  bool merge_families_;
+  std::vector<std::unique_ptr<Family<Counter>>> counters_;
```

Comment:
@jupp0r do you have a better idea how to store the not-type erased families w/o using std::variant and w/o using an ugly downcast?

Reply:
With C++17 it looks pretty OK: 71d9a18b6577131b3c69c2b173d2287db1a42ddf

Reply:
I don't think this is too bad, actually.